server/cli/src/main/java/org/elasticsearch/cli/MultiCommand.java

```diff
@@ -74,4 +75,16 @@ protected void execute(Terminal terminal, OptionSet options) throws Exception {
         }
         subcommand.mainWithoutErrorHandling(Arrays.copyOfRange(args, 1, args.length), terminal);
     }
+
+    @Override
+    public void close() throws IOException {
+        if (subcommands.isEmpty()) {
```

Comment:
I don't think `close()` is the place to be erroring if subcommands were never setup. This would already happen in `execute`.